### DISCUSSION TITLE
Boundary distort the field in refinement.

### DISCUSSION THREAD

**Discussion by jiecaix** [BLAST-WarpX/warpx#4517] [Q&A] [answered]
I'm using the refinement mesh in simulation, and i find some problem in it.
1. I set a small laser pulse at Level 1 region, and the field is indeed a good small laser pulse in level 1 region. But the level 0 field with the low frequency projection of level 1 laser field will gradually expand longer and longer and shift back.
### Field of Level 0 at end of timestep
![image](https://github.com/ECP-WarpX/WarpX/assets/56915687/7aa2d267-8a96-44d1-9c6f-cb1d16562e3a)

2. The level 1 window following a field look like Numerical Cherenkov radiation.
### Field of Level 0 at end of timestep
![image](https://github.com/ECP-WarpX/WarpX/assets/56915687/1a30747d-ebd4-4e65-b49f-e80369821b54)

3. Near the boundary of level 1, the FieldProbe module will probe a strong field compared to others region.
### Field probe by FieldProbe
![image](https://github.com/ECP-WarpX/WarpX/assets/56915687/48ee10d6-4684-4ed2-b002-085f57d0c5bf)

**Comment by ax3l** [comment, marked as answer]
@zhazhajust it is a bit hard to follow here, but you might want to post more details and try higher resolution and/or other field solvers.

There is also an option `lasers.deposit_on_main_grid` that you can try:
https://warpx.readthedocs.io/en/latest/usage/parameters.html#laser-initialization
